src/app/pages/categories/management/pages-categories-management.component.ts

```diff
@@ -53,6 +55,29 @@ export class PagesCategoriesManagementComponent implements OnInit, OnDestroy {
     this.categoriesTypeSub.unsubscribe();
   }
 
+  public handleCreateCategory(type: WalletTransactionType): void {
+    this.pagesCategoriesEditorService
+      .openEditor(new TransactionCategory({ id: null, name: '', type }))
```

Comment:
You can delegate creating empty category to the service. So category argument for `openEditor` can be optional and if not provided you can create in that way empty category in `openEditor`